### PR TITLE
#Task 238575:[Java][SDP] Prepare script to convert Merf to SDP for do…

### PR DESCRIPTION
**Pull request by v-haiboz**
…xygen pipeline result (doxygen)

Create DataContracts

### CODE REVIEW

**Review by TianqiZhang — APPROVED** [review]

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ItemType.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        Enum,
+        Field,
+        Property,
+        Event,
+        Constructor,
+        Method,
+        Operator,
+        Container,
+        AttachedEvent,
+        AttachedProperty
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ItemType.cs]: I think we do not have "Assembly", "Struct", "Operator", "AttachedEvent", "AttachedProperty"

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Shared/EnumMethod.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+﻿namespace Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Shared/EnumMethod.cs]: could EnumMethod share some model with Member?

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ConfigModel.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    public class ConfigModel
+    {
+        [JsonProperty(Constants.InputPaths, Required = Required.DisallowNull)]
+        public List<string> InputPaths { get; set; }
+
+        [JsonProperty(Constants.OutputPath, Required = Required.DisallowNull)]
+        public string OutputPath { get; set; }
+
+        [JsonProperty(Constants.Language)]
+        public string Language { get; set; } = "java";
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ConfigModel.cs]: why do we need "Language" here?

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/MemberType.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    [Serializable]
+    public enum MemberType
+    {
+        Namespace,
+        Class,
+        Struct,
+        Enum,
+        Field,
+        Method,//MemberFunction
+        //Friend,
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/MemberType.cs]: Friend?

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Shared/EnumMethod.cs (hunk 2 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Shared/EnumMethod.cs]: Member base on EnumMethod?

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/ArticleItemYaml.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
+﻿namespace Microsoft.Content.Build.MerfToSDPForJava.DataContracts
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/ArticleItemYaml.cs]: suggest we put Merf data models and SDP data models to different folder which will be easier identify

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ItemType.cs (hunk 1 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ItemType.cs]: sure

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/ArticleItemYaml.cs (hunk 5 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/ArticleItemYaml.cs]: sure

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/PageModel.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    public class PageModel
+    {
+        [YamlMember(Alias = "items")]
+        public List<ArticleItemYaml> Items { get; set; } = new List<ArticleItemYaml>();
+
+        [YamlMember(Alias = "metadata")]
+        public Dictionary<string, object> Metadata { get; set; }
+
+        [YamlMember(Alias = "references")]
+        public List<ReferenceViewModel> References { get; set; } = new List<ReferenceViewModel>();
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/PageModel.cs]: if we do not need the information in "References", maybe it is not necessary to deserialize it.

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/Hierarchy/HierarchyType.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Microsoft.Content.Build.MerfToSDPForJava.DataContracts
+{
+    public enum HierarchyType
+    {
+        Namespace,
+        Class,
+        Struct,
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/Hierarchy/HierarchyType.cs]: there is no "Struct"

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/Hierarchy/HierarchyChange.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
+﻿namespace Microsoft.Content.Build.MerfToSDPForJava.DataContracts
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/Hierarchy/HierarchyChange.cs]: why do we need to "HierarchyChange" for please?

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ServiceMapping/ServiceMapping.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Microsoft.Content.Build.MerfToSDPForJava.DataContracts
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ServiceMapping/ServiceMapping.cs]: how are we going to use "ServiceMapping" please?

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ItemType.cs (hunk 1 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ItemType.cs]: Updated

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Shared/EnumMethod.cs (hunk 2 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Shared/EnumMethod.cs]: Updated

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ConfigModel.cs (hunk 3 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/ConfigModel.cs]: Remove

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/MemberType.cs (hunk 4 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/MemberType.cs]: Remove

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/ArticleItemYaml.cs (hunk 5 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/ArticleItemYaml.cs]: Updated

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/PageModel.cs (hunk 6 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ViewModel/PageModel.cs]: Removed

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ServiceMapping/ServiceMapping.cs (hunk 9 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/ServiceMapping/ServiceMapping.cs]: Remove

File: src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Merf/ArticleItemYaml.cs [hunk 10]
```diff
@@ -0,0 +1,109 @@
+﻿namespace Microsoft.Content.Build.MerfToSDPForJava.DataContracts.Merf
```
Comment (anmeng10101) [inline src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Merf/ArticleItemYaml.cs]: let's use full name "ManagedReference" instead of "Merf"

Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.SDP.DataContracts/Merf/ArticleItemYaml.cs]: updated

File: src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/Hierarchy/HierarchyChange.cs (hunk 8 above)
Comment (v-haiboz) [reply src/Microsoft.Content.Build.MerfToSDPForJava.DataContracts/Hierarchy/HierarchyChange.cs]: updated

**Pull request merged**